src/components/AegisActions/AegisActions.vue

```diff
@@ -0,0 +1,181 @@
+<script setup lang="ts">
+import { ref, watch } from 'vue';
+
+import { osimRuntime } from '@/stores/osimRuntime';
+
+const props = defineProps<{
+  canShowFeedback: boolean;
+  canSuggest: boolean;
+  hasAppliedSuggestion: boolean;
+  hasMultipleSuggestions: boolean;
+  isSuggesting: boolean;
+  selectedIndex: number;
+  suggestions: string[];
+  tooltipText: string;
+}>();
+
+const emit = defineEmits<{
+  feedback: ['negative' | 'positive'];
+  revert: [];
+  selectSuggestion: [index: number];
+  suggest: [];
+}>();
+
+const showDropdown = ref(false);
+
+watch(() => props.isSuggesting, newValue => newValue && (showDropdown.value = false));
+
+function toggleDropdown() {
+  showDropdown.value = !showDropdown.value;
+}
+
+function selectSuggestion(index: number) {
+  emit('selectSuggestion', index);
+  showDropdown.value = false;
+}
+
+function handleDropdownBlur(event: FocusEvent) {
+  const currentTarget = event.currentTarget as HTMLElement;
+  const relatedTarget = event.relatedTarget as HTMLElement;
+
+  if (!currentTarget.contains(relatedTarget)) {
+    showDropdown.value = false;
+  }
+}
+</script>
+
+<template>
+  <div v-if="osimRuntime.flags?.aiCweSuggestions === true" class="d-flex align-items-center">
+    <i
+      class="bi-stars label-icon"
+      :class="{ disabled: !canSuggest, applied: hasAppliedSuggestion }"
+      :title="tooltipText"
+      @click.prevent.stop="canSuggest && emit('suggest')"
+    />
+    <div
+      v-if="hasMultipleSuggestions && !isSuggesting"
+      class="position-relative ms-1"
+      tabindex="0"
+      @focusout="handleDropdownBlur"
+    >
+      <i
+        class="bi-chevron-down label-icon dropdown-arrow"
+        :class="{ 'rotated': showDropdown }"
+        title="Select from multiple suggestions"
+        @click.prevent.stop="toggleDropdown"
+      />
+      <div
+        v-if="showDropdown"
+        class="suggestion-dropdown position-absolute bg-white border rounded shadow-sm"
+      >
+        <div class="dropdown-header px-2 py-1 bg-light border-bottom">
```

Comment:
Bootsrap dropdowns have dividers https://getbootstrap.com/docs/5.3/components/dropdowns/#dividers in case you want to take a look if you like it more than adding a border